backend/infrahub/lock.py

```diff
@@ -245,12 +248,48 @@ async def local_schema_wait(self) -> None:
         await self.get(name=LOCAL_SCHEMA_LOCK).event.wait()
 
     def global_schema_lock(self) -> InfrahubMultiLock:
-        return InfrahubMultiLock(_registry=self, locks=[LOCAL_SCHEMA_LOCK, GLOBAL_SCHEMA_LOCK])
+        return InfrahubMultiLock(lock_registry=self, locks=[LOCAL_SCHEMA_LOCK, GLOBAL_SCHEMA_LOCK])
 
     def global_graph_lock(self) -> InfrahubMultiLock:
-        return InfrahubMultiLock(_registry=self, locks=[LOCAL_SCHEMA_LOCK, GLOBAL_GRAPH_LOCK, GLOBAL_SCHEMA_LOCK])
+        return InfrahubMultiLock(lock_registry=self, locks=[LOCAL_SCHEMA_LOCK, GLOBAL_GRAPH_LOCK, GLOBAL_SCHEMA_LOCK])
 
 
 def initialize_lock(local_only: bool = False, service: Optional[InfrahubServices] = None) -> None:
     global registry  # pylint: disable=global-statement
     registry = InfrahubLockRegistry(local_only=local_only, service=service)
+
+
+def build_object_lock_name(name: str) -> str:
+    return f"global.object.{name}"
+
+
+def get_kinds_to_lock_on_object_mutation(node_schema: GeneratedBaseNodeSchema) -> list[str]:
```

Comment:
I think we should move this function to the SchemaBranch object directly to avoid relying on the global registry.
Also in the current implementation, the branch is missing so it will always pull the schema for the main branch